test/Microsoft.AspNetCore.Mvc.Core.Test/Internal/ControllerActionInvokerCacheTest.cs

```diff
@@ -186,8 +186,80 @@ public void GetControllerActionMethodExecutor_CachesExecutor()
             Assert.Same(executor1, executor2);
         }
 
+        [Theory]
```

Comment:
This test covers the scenario. The filter provider adds a new filter item always to the result. If there were duplicate items the count of item would be incorrect.